### PR TITLE
Turn washing filthy clothes into long action

### LINKED ISSUE #20468: Being attacked while washing filthy clothes
So, I almost died while I was near a swamp, washing a Military rucksack (or so). I was attacked by a pack of dogs, and heard a strike or two of a nearby shocker zombie.

I guess it was possible to stop the activity by hitting ".", but it was so quick, I couldn't do anything.

Just like butchering, it could use a warning like "you see _monster_ aproaching".

### PR DESCRIPTION
**Pull request by Night-Pryanik**
Closes #20468.

### CODE REVIEW

File: src/savegame_legacy.cpp [hunk 1]
```diff
@@ -538,6 +538,7 @@ void player_activity::deserialize_legacy_type( int legacy_type, activity_id &des
         activity_id( "ACT_WAIT_NPC" ),
         activity_id( "ACT_CLEAR_RUBBLE" ),
         activity_id( "ACT_MEDITATE" ),
+        activity_id( "ACT_WASH" ),
```
Comment (BevapDin) [inline src/savegame_legacy.cpp]: Please don't add anything to legacy things like this. They exist only to convert saves in old format to the current format.

File: src/activity_handlers.cpp [hunk 2]
```diff
@@ -1946,3 +1947,18 @@ void activity_handlers::aim_finish( player_activity *, player * )
     // you only re-enter if it gets set again.
     return;
 }
+
+void activity_handlers::washing_finish( player_activity *act, player *p )
+{
+    item &filthy_item = p->i_at(act->position);
+
+    if( p->is_worn( filthy_item ) ) {
+        filthy_item.on_takeoff( g->u );
```
Comment (BevapDin) [inline src/activity_handlers.cpp]: This (and the line below) should use `*p` instead of `g->u`.

File: src/savegame_legacy.cpp (hunk 1 above)
Comment (Night-Pryanik) [inline src/savegame_legacy.cpp]: Ok, noted. Removed my addition.

File: src/activity_handlers.cpp (hunk 2 above)
Comment (Night-Pryanik) [inline src/activity_handlers.cpp]: Changed.

**Comment by kevingranade** [comment]
Had to do a rebase instead of merge, so it didn't auto-close.

**Pull request closed without merging**